core/src/test/java/io/grpc/internal/DelayedStreamTest.java

```diff
@@ -154,11 +155,15 @@ public void setStream_cantCreateTwice() {
 
   @Test
   public void streamCancelled() {
-    stream.cancel(Status.CANCELLED);
+    Status status = Status.CANCELLED.withDescription("unique status message");
+    stream.cancel(status);
+    verify(listener).closed(same(status), isA(Metadata.class));
+    verifyNoMoreInteractions(listener);
 
-    // Should be a no op, and not fail due to transport not returning a newStream
+    // Should be _mostly_ a no op, and not fail due to transport not returning a newStream
     stream.setStream(realStream);
 
-    verify(listener).closed(eq(Status.CANCELLED), isA(Metadata.class));
+    verifyNoMoreInteractions(listener);
```

Comment:
Actually, it was trying to verify something that was good. And in fact, now I'm pretty certain there is a bug here because `listener.closed` _will_ be called twice. The test didn't notice because stream is a mock.

Apart from the bug, would you feel better if line 161 was a reset(listener) instead? That does seem more appropriate.